source/ota.c

```diff
@@ -3113,26 +3110,36 @@ OtaErr_t OTA_Init( OtaAppBuffer_t * pOtaBuffer,
          */
         ( void ) otaAgent.pOtaInterface->os.event.init( NULL );
 
-        if( pThingName == NULL )
+        if( pOtaBuffer == NULL )
         {
-            LogError( ( "Error: Thing name is NULL.\r\n" ) );
+            LogError( ( "Error: pOtaBuffer is NULL.\r\n" ) );
         }
         else
         {
-            uint32_t strLength = ( uint32_t ) ( strlen( ( const char * ) pThingName ) );
+            /* Initialize application buffers. */
+            initializeAppBuffers( pOtaBuffer );
 
-            if( strLength <= otaconfigMAX_THINGNAME_LEN )
+            if( pThingName == NULL )
             {
-                /*
-                 * Store the Thing name to be used for topics later. Include zero terminator
-                 * when saving the Thing name.
-                 */
-                ( void ) memcpy( otaAgent.pThingName, pThingName, strLength + 1UL );
-                returnStatus = OtaErrNone;
+                LogError( ( "Error: Thing name is NULL.\r\n" ) );
             }
             else
             {
-                LogError( ( "Error: Thing name is too long.\r\n" ) );
+                uint32_t strLength = ( uint32_t ) ( strlen( ( const char * ) pThingName ) );
+
+                if( strLength <= otaconfigMAX_THINGNAME_LEN )
```

Comment:
This could be simplified to
```
if( strnlen( pThingName, otaconfigMAX_THINGNAME_LEN + 1 ) <= otaconfigMAX_THINGNAME_LEN )
{
    (void) strncpy(  otaAgent.pThingName, pThingName, otaconfigMAX_THINGNAME_LEN );
}
else
{
    LogError("...");
    returnStatus = xxx;
}